src/mlpack/tests/string_utility_test.cpp

```diff
@@ -0,0 +1,178 @@
+      boost::hash<boost::string_view>> maps = en.Mappings();
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Test for CharSplit class.
+*/
+BOOST_AUTO_TEST_CASE(CharSplitTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  en.Encode(arr, output, CharSplit(" "));
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>> maps = en.Mappings();
+  // Checking that everything is mapped to different numbers.
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mappend only once.
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Test Dictionary encoding for characters using lamda function.
+*/
+BOOST_AUTO_TEST_CASE(DictionaryEncodingCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "GACCA";
+  arr[1] = "ABCABCD";
+  arr[2] = "GAB";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  // Passing a empty string to encode characters
+  en.Encode(arr, output, [](boost::string_view& str) {
+      if (str.empty())
+        return str;
+      boost::string_view retval = str.substr(0, 1);
+      str.remove_prefix(1);
+      return retval;
+  });
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>> maps = en.Mappings();
+  // Checking that everything is mapped to different numbers.
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mappend only once.
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Testing the functionality of copy constructor.
+*/
+BOOST_AUTO_TEST_CASE(CopyConstructorCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  en.Encode(arr, output, CharSplit(" "));
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>> maps = en.Mappings();
+  const std::deque<std::string>& original = en.OriginalStrings();
+  data::DictionaryEncoding en2 = en;
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>> maps2 = en2.Mappings();
+  const std::deque<std::string>& original2 = en2.OriginalStrings();
+  // Comparing both of them.
+  std::deque<std::string>::const_iterator jt = original2.begin();
+  for (auto it = original.begin(); it != original.end(); it++)
+  {
+    BOOST_REQUIRE_EQUAL(maps2.at(*jt), maps.at(*it));
```

Comment:
I am not sure, does `std::unordered_map` support the comparison operator `==`?

Reply:
Made apt change